docs/maven-plugin/pom.xml

```diff
@@ -56,7 +56,7 @@
         </dependency>
         <dependency>
             <groupId>org.keycloak</groupId>
-            <artifactId>keycloak-quarkus-server</artifactId>
+            <artifactId>keycloak-quarkus-server-deployment</artifactId>
```

Comment:
Just out of curiosity, why do docs need to depend on build-time part of server? It does not contain anything relevant to generating config docs AFAIK.

Reply:
If I only specify `keycloak-quarkus-server`, it fails with the same error I get when building the `dist`. 

The culprit might be somewhere in the Quarkus maven plugin that wants to get some information from the deployment. So it must be somewhere in Maven's reactor (or whatever that is called) to work. You can't build ust the `quarkus-server` without also building the `quarkus-server-deployment` when using `also-make`. Strange ... it feels like a workaround, but I currently see no other way to fix this.

https://github.com/keycloak/keycloak/blob/7c013e8d48cc8ce1a2543ba0b036eb99c2a2633b/quarkus/runtime/pom.xml#L688-L702

Reply:
Yeah, that is a good point. I wonder what was the reasoning for not including `keycloak-quarkus-server-deployment` as a dependency in `keycloak-quarkus-server` since technically it is its dependency.

@pedroigor @mabartos Do you perhaps have more context?

Reply:
> Do you perhaps have more context?

Ah, [here](https://quarkus.io/guides/writing-extensions#project-setup)'s the reasoning:
```
Under no circumstances can the runtime module depend on a deployment artifact. This would result in pulling all the deployment time code into runtime scope, which defeats the purpose of having the split.
```

Reply:
Still it's some kind cyclic dependency on how `quarkus-extension-maven-plugin` works. 

So this PR is the way to go?

Reply:
Yes, as far as I can tell, it's the way to go. The build dist also looks identical.